e2e/scenarios/grantDelegation.ethereum.test.ts

```diff
@@ -0,0 +1,183 @@
+import { getFundingSource } from '../scaffolding/funding';
+
+const fundingSource = getFundingSource('grant-delegation-ethereum');
+
+describe('Delegation Scenario Tests Ethereum', function () {
+  let keys: KeyringPair;
+  let otherMsaKeys: KeyringPair;
+  let thirdMsaKeys: KeyringPair;
+  let noMsaKeys: KeyringPair;
+  let providerKeys: KeyringPair;
+  let otherProviderKeys: KeyringPair;
+  let schemaId: u16;
+  let schemaId2: SchemaId;
+  let providerId: u64;
+  let otherProviderId: u64;
+  let msaId: u64;
+  let otherMsaId: u64;
+  let thirdMsaId: u64;
+
+  before(async function () {
+    // Fund all the different keys
+    [noMsaKeys, keys, otherMsaKeys, thirdMsaKeys, providerKeys, otherProviderKeys] = await createAndFundKeypairs(
+      fundingSource,
+      ['noMsaKeys', 'keys', 'otherMsaKeys', 'thirdMsaKeys', 'providerKeys', 'otherProviderKeys'],
+      1n * DOLLARS,
+      'ethereum'
+    );
+
+    const { target: msaCreatedEvent1 } = await ExtrinsicHelper.createMsa(keys).signAndSend();
+    msaId = msaCreatedEvent1!.data.msaId;
+
+    const { target: msaCreatedEvent2 } = await ExtrinsicHelper.createMsa(otherMsaKeys).signAndSend();
+    otherMsaId = msaCreatedEvent2!.data.msaId;
+
+    const { target: msaCreatedEvent3 } = await ExtrinsicHelper.createMsa(thirdMsaKeys).signAndSend();
+    thirdMsaId = msaCreatedEvent3!.data.msaId;
+
+    let createProviderMsaOp = ExtrinsicHelper.createMsa(providerKeys);
+    await createProviderMsaOp.signAndSend();
+    let createProviderOp = ExtrinsicHelper.createProvider(providerKeys, 'MyPoster');
+    let { target: providerEvent } = await createProviderOp.signAndSend();
+    assert.notEqual(providerEvent, undefined, 'setup should return a ProviderCreated event');
+    providerId = providerEvent!.data.providerId;
+
+    createProviderMsaOp = ExtrinsicHelper.createMsa(otherProviderKeys);
+    await createProviderMsaOp.signAndSend();
+    createProviderOp = ExtrinsicHelper.createProvider(otherProviderKeys, 'MyPoster');
+    ({ target: providerEvent } = await createProviderOp.signAndSend());
+    assert.notEqual(providerEvent, undefined, 'setup should return a ProviderCreated event');
+    otherProviderId = providerEvent!.data.providerId;
+
+    const schema = {
+      type: 'record',
+      name: 'Post',
+      fields: [
+        { name: 'title', type: { name: 'Title', type: 'string' } },
+        { name: 'content', type: { name: 'Content', type: 'string' } },
+        { name: 'fromId', type: { name: 'DSNPId', type: 'fixed', size: 8 } },
+        { name: 'objectId', type: 'DSNPId' },
+      ],
+    };
+
+    schemaId = await ExtrinsicHelper.getOrCreateSchemaV3(
+      keys,
+      schema,
+      'AvroBinary',
+      'OnChain',
+      [],
+      'test.grantDelegation'
+    );
+
+    schemaId2 = await ExtrinsicHelper.getOrCreateSchemaV3(
+      keys,
+      schema,
+      'AvroBinary',
+      'OnChain',
+      [],
+      'test.grantDelegationSecond'
+    );
+  });
```

Comment:
nit: probably don't need to create schemas any more, unless actually testing the schemas pallet, as mainnet schemas should be available on all chains now

Reply:
yeah this is getting the schemaId if it already exists but I agree that we can refactor all these schema creation everywhere and simplify it